@here/harp-vectortile-datasource/lib/VectorTileRestClient.ts

```diff
@@ -0,0 +1,465 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import "@here/harp-fetch";
+import { TileKey, TilingScheme } from "@here/harp-geoutils";
+import { DataProvider } from "@here/harp-mapview-decoder";
+import { TransferManager } from "@here/harp-transfer-manager";
+import { LoggerManager } from "@here/harp-utils";
+
+const logger = LoggerManager.instance.create("VectorTileRestClient");
+
+// tslint:disable:max-line-length
+export enum APIFormat {
+    /**
+     * Use the REST API format of HERE Vector Tiles Server component version 1.
+     *
+     * Usage:
+     *
+     *     <OmvRestClientParams.baseUrl>/<zoom>/<X>/<Y>/omv
+     *
+     * If [[VectorTileRestClientParams.authenticationToken]] is provided, it will be added as HTTP header:
+     *
+     *     Authorization: Bearer $authenticationToken
+     *
+     * Format definition:
+     * `//http|s://<base-url>/{API version}/{layers}/{projection}/{z}/{x}/{y}/{format}`
+     *
+     * Default authentication method used: [[AuthenticationTypeBearer]].
+     */
+    HereV1,
+
+    /**
+     * Use the REST API format of Mapbox Vector Tile API v4.
+     *
+     * Usage:
+     * `<VectorTileRestClientParams.baseUrl>/<zoom>/<X>/<Y>.mvt?access_token=<VectorTileRestClientParams.authenticationCode>`
+     *
+     * Format definition:
+     * `http|s://<base-url>/v4/{map_id}/{z}/{x}/{y}{@2x}.{format}?[style]&access_token={access_token}`
+     *
+     * Sample URL:
+     * `http://a.tiles.mapbox.com/v4/mapbox.mapbox-streets-v7/14/4823/6160.mvt?access_token=your-mapbox-access-token`
+     *
+     * Default authentication method used: [[AuthenticationTypeAccessToken]].
+     */
+    MapboxV4,
+
+    /**
+     * Use the REST API format of XYZ Vector Tile API in MVT format.
+     *
+     * Usage:
+     * `<VectorTileRestClientParams.baseUrl>/tiles/omsbase/256/<zoom>/<X>/<Y>.mvt?access_token=<VectorTileRestClientParams.authenticationCode>`
+     *
+     * Format definition:
+     * `http|s://<base-url>/tiles/{layers}/{z}/{x}/{y}/{format}?access_token={access_token}`
+     *
+     * Sample URL:
+     * `https://xyz.api.here.com/tiles/osmbase/256/all/16/19293/24641.mvt?access_token=your-xyz-access-token`
+     *
+     * Default authentication method used: [[AuthenticationTypeAccessToken]].
+     */
+    XYZMVT,
+
+    /**
+     * Use the REST API format of XYZ Vector Tile API in JSON format.
+     *
+     * Usage:
+     * `<VectorTileRestClientParams.baseUrl>/tiles/omsbase/256/<zoom>/<X>/<Y>.mvt?access_token=<VectorTileRestClientParams.authenticationCode>`
+     *
+     * Format definition:
+     * `http|s://<base-url>/tiles/{layers}/{z}/{x}/{y}/{format}?access_token={access_token}`
+     *
+     * Sample URL:
+     * `https://xyz.api.here.com/tiles/osmbase/256/all/16/19293/24641.json?access_token=your-xyz-api-key`
+     *
+     * Default authentication method used: [[AuthenticationTypeAccessToken]].
+     */
+    XYZJson,
+
+    /**
+     * Use the REST API format of XYZ Vector Tile API in OMV format.
```

Comment:
Would it be worth adding a new format and deprecating XYZOMV

Reply:
Yes that's the topic of the discussion to have tomorrow